index.js

```diff
@@ -801,25 +801,27 @@
   //# Maybe :: TypeRep Maybe
   //.
   //. The [type representative](#type-representatives) for the Maybe type.
-  var Maybe = S.Maybe = function Maybe() {
-    if (arguments[0] !== sentinel) {
-      throw new Error('Cannot instantiate Maybe');
-    }
+  var Maybe = S.Maybe = function Maybe(x, box) {
+    if (x !== sentinel) throw new Error('Cannot instantiate Maybe');
+    var isJust = box.length > 0;
+    if (isJust) this.value = box[0];
+    this.isNothing = !isJust;
+    this.isJust = isJust;
   };
 
   //# Maybe.empty :: -> Maybe a
   //.
-  //. Returns a Nothing.
+  //. Returns Nothing.
```

Comment:
I :heart: getting to remove “a” here: conceptually there's a single `Nothing` value.